src/main/java/com/google/api/codegen/DiscoveryContext.java

```diff
@@ -84,6 +84,108 @@ public String getSimpleName(String name) {
     return name.substring(name.lastIndexOf('.') + 1);
   }
 
+  public String getSampleVarName(String typeName) {
+    return upperCamelToLowerCamel(getSimpleName(typeName));
```

Comment:
The latter is correct, though as the only current override is for C#, this may be refactored later. Will add comment for now.